### ISSUE TITLE
[ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:

### ISSUE THREAD

**Issue by PonnamKarthik** [renefloor/flutter_cached_network_image#36]
```
[ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
'package:flutter/src/painting/image_stream.dart': Failed assertion: line 362 pos 12: '_codec != null': is not true.
#0      _AssertionError._doThrowNew (dart:core/runtime/liberrors_patch.dart:37:39)
#1      _AssertionError._throwNew (dart:core/runtime/liberrors_patch.dart:33:5)
#2      MultiFrameImageStreamCompleter._handleCodecReady (package:flutter/src/painting/image_stream.dart)
#3      _RootZone.runUnary (dart:async/zone.dart:1381:54)
#4      _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#5      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:633:45)
#6      Future._propagateToListeners (dart:async/future_impl.dart:662:32)
#7      Future._completeError (dart:async/future_impl.dart:485:5)
#8      _SyncCompleter._completeError (dart:async/future_impl.dart:55:12)
#9      _Completer.completeError (dart:async/future_impl.dart:27:5)
#10     CachedNetworkImageProvider._loadAsync (package:cached_network_image/cached_network_image.dart)
<asynchronous suspension>
#11     CachedNetworkImageProvider.load (package:cached_network_image/cached_network_image.dart:462:16)
#12     ImageProvider.resolve.<anonymous closure>.<anonymous closure> (package:flutter/src/painting/image_provider.dart:265:86)
#13     ImageCache.putIfAbsent (package:flutter/src/painting/image_cache.dart:82:22)
#14     ImageProvider.resolve.<anonymous closure> (package:flutter/src/painting/image_provider.dart:265:63)
#15     SynchronousFuture.then (package:flutter/src/foundation/synchronous_future.dart:38:29)
#16     ImageProvider.resolve (package:flutter/src/painting/image_provider.dart:263:30)
#17     _ImageProviderResolver.resolve (package:cached_network_image/cached_network_image.dart:193:29)
#18     _CachedNetworkImageState._resolveImage (package:cached_network_image/cached_network_image.dart:284:20)
#19     _CachedNetworkImageState.didChangeDependencies (package:cached_network_image/cached_network_image.dart:261:5)
#20     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3750:12)
#21     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
#22     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#23     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#24     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4637:14)
#25     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#26     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#27     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4637:14)
#28     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#29     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#30     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636:16)
#31     Element.rebuild (package:flutter/src/widgets/framework.dart:3478:5)
#32     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605:5)
#33     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
#34     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#35     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4742:32)
#36     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#37     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#38     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636:16)
#39     Element.rebuild (package:flutter/src/widgets/framework.dart:3478:5)
#40     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605:5)
#41     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
#42     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#43     Element.updateChild (package:flutter/src/widgets/f
[ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
NoSuchMethodError: The getter '_value' was called on null.
Receiver: null
Tried calling: _value
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:46:5)
#1      DateTime.difference (dart:core/runtime/libdate_patch.dart:183:54)
#2      CacheManager._cleanCache (package:flutter_cache_manager/flutter_cache_manager.dart:129:45)
<asynchronous suspension>
#3      CacheManager._save (package:flutter_cache_manager/flutter_cache_manager.dart:75:11)
<asynchronous suspension>
#4      CacheManager.getFile (package:flutter_cache_manager/flutter_cache_manager.dart:243:5)
<asynchronous suspension>
#5      CachedNetworkImageProvider._loadAsync (package:cached_network_image/cached_network_image.dart:474:35)
<asynchronous suspension>
#6      CachedNetworkImageProvider.load (package:cached_network_image/cached_network_image.dart:462:16)
#7      ImageProvider.resolve.<anonymous closure>.<anonymous closure> (package:flutter/src/painting/image_provider.dart:265:86)
#8      ImageCache.putIfAbsent (package:flutter/src/painting/image_cache.dart:82:22)
#9      ImageProvider.resolve.<anonymous closure> (package:flutter/src/painting/image_provider.dart:265:63)
#10     SynchronousFuture.then (package:flutter/src/foundation/synchronous_future.dart:38:29)
#11     ImageProvider.resolve (package:flutter/src/painting/image_provider.dart:263:30)
#12     _ImageProviderResolver.resolve (package:cached_network_image/cached_network_image.dart:193:29)
#13     _CachedNetworkImageState._resolveImage (package:cached_network_image/cached_network_image.dart:284:20)
#14     _CachedNetworkImageState.didChangeDependencies (package:cached_network_image/cached_network_image.dart:261:5)
#15     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3750:12)
#16     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
#17     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#18     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#19     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4637:14)
#20     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#21     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#22     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4637:14)
#23     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#24     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#25     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636:16)
#26     Element.rebuild (package:flutter/src/widgets/framework.dart:3478:5)
#27     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605:5)
#28     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
#29     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#30     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4742:32)
#31     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#32     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#33     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3636:16)
#34     Element.rebuild (package:flutter/src/widgets/framework.dart:3478:5)
#35     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3605:5)
#36     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
#37     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#38     Element.updateChild (package:flutter/src/widgets/framework.dart:2693:12)
#39     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4637:14)
#40     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2890:14)
#41     Element.updateChild (package:flutter/src/widgets/framework.dart:26
```

**Comment by renefloor** [comment]
Relevant part of the error:
```
NoSuchMethodError: The getter '_value' was called on null.
Receiver: null
Tried calling: _value
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:46:5)
#1      DateTime.difference (dart:core/runtime/libdate_patch.dart:183:54)
#2      CacheManager._cleanCache (package:flutter_cache_manager/flutter_cache_manager.dart:129:45)
<asynchronous suspension>
#3      CacheManager._save (package:flutter_cache_manager/flutter_cache_manager.dart:75:11)
<asynchronous suspension>
#4      CacheManager.getFile (package:flutter_cache_manager/flutter_cache_manager.dart:243:5)
<asynchronous suspension>
```

**Comment by PonnamKarthik** [comment]
what is the fix ?

**Comment by renefloor** [comment]
Not sure yet. I'll look into this when I have time.

**Comment by renefloor** [comment]
Can you check if your problems are fixed in 0.4.1?
